src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -957,15 +951,23 @@ private void CheckRestrictedTypeReceiver(BoundExpression expression, Compilation
             {
                 if (analyzedArguments.HasErrors)
                 {
-                    // Errors for arguments have already been reported, except for unbound lambdas.
+                    // Errors for arguments have already been reported, except for unbound lambdas and switch expressions.
                     // We report those now.
                     foreach (var argument in analyzedArguments.Arguments)
                     {
-                        var unboundLambda = argument as UnboundLambda;
-                        if (unboundLambda != null)
+                        switch (argument)
                         {
-                            var boundWithErrors = unboundLambda.BindForErrorRecovery();
-                            diagnostics.AddRange(boundWithErrors.Diagnostics);
+                            case UnboundLambda unboundLambda:
+                                var boundWithErrors = unboundLambda.BindForErrorRecovery();
+                                diagnostics.AddRange(boundWithErrors.Diagnostics);
+                                break;
+                            case BoundTupleLiteral _:
```

Comment:
>_ [](start = 51, length = 1)

nit: you could just not put anything here at all

Reply:
That's not the way pattern-matching works.  A type is not a pattern.

---
In reply to: [301339843](https://github.com/dotnet/roslyn/pull/37052#discussion_r301339843) [](ancestors = 301339843)